tests/rados/test_osd_crashes.py

```diff
@@ -0,0 +1,212 @@
+            "This test is to verify whether OSDs deployed with NCB allocator file"
+            "face corruption during subsequent restarts and upgrades \n"
+            "1. Create a pool with single PG \n"
+            "2. Obtain the acting set of the created pool \n"
+            "3. Re-deploy the OSDs part of acting set after faking their underlying disk as non-rotational \n"
+            "4. Check the current value of bluestore_allocation_from_file \n"
+            "5. Check the current value of osd_fast_shutdown \n"
+            "6. Check existence of Allocator NCB file in Bluefs files List \n"
+            "7. Use fragmentation python script to write first set of fragmented objects \n"
+            "8. Stop primary OSD using systemctl and verify that the last "
+            "file saved during shutdown is allocator file(ino 27) \n"
+            "9. Use fragmentation python script to write second set of fragmented objects \n"
+            "10. Stop primary OSD using systemctl and verify the "
+            "increase in number of extents for allocator file(ino 27) \n"
+            "11. Use fragmentation python script to write third set of fragmented objects \n"
+            "12. Stop primary OSD using systemctl and verify the "
+            "increase in number of extents for allocator file(ino 27) \n"
+            "13. Remove around 10K fragmented objects  at different offsets \n"
+            "14. Stop primary OSD using systemctl and verify the "
+            "increase in number of extents for allocator file(ino 27) \n"
+            "15. Upgrade the cluster and ensure cluster Health is OK \n"
+        )
+
+        log.info(desc)
+        _config = config.get("verify_allocator_corruption")
+        pool_cfg = config.get("pool_config")
+        pool_name = pool_cfg["pool_name"]
+        log.info(
+            "\n \n Running test to verify OSD recovery in case of Allocator File corruption"
+        )
+
+        try:
+            if _config.get("issue_reproduction"):
+                # create pool with given config
+                assert rados_obj.create_pool(**pool_cfg), "Pool creation failed"
+
+                # get acting set of created pool
+                acting_set = rados_obj.get_pg_acting_set(pool_name=pool_name)
+                prim_osd = acting_set[0]
+
+                # for each OSD in acting set, change the underlying disk type
+                for _osd in acting_set:
+                    rados_obj.switch_osd_device_type(osd_id=_osd, rota_val=0)
+                time.sleep(5)
+
+                assert mon_obj.show_config(
+                    daemon="osd", id=prim_osd, param="bluestore_allocation_from_file"
+                ), "bluestore_allocation_from_file is not true"
+                assert mon_obj.show_config(
+                    daemon="osd", id=prim_osd, param="osd_fast_shutdown"
+                ), "osd_fast_shutdown is not true"
+
+                # check for existence of NCB allocator file
```

Comment:
Add the comment that the file is only generated for SSD based devices

Reply:
added